src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferConstCharOverConstUnitStringAnalyzer.cs

```diff
@@ -0,0 +1,116 @@
+                                                                                      DiagnosticCategory.Performance,
+                                                                                      RuleLevel.IdeSuggestion,
+                                                                                      s_localizableDescription,
+                                                                                      isPortedFxCopRule: false,
+                                                                                      isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.RegisterOperationAction(AnalyzeSymbol, OperationKind.Invocation);
+        }
+
+        private static void AnalyzeSymbol(OperationAnalysisContext context)
+        {
+            IInvocationOperation invocationOperation = (IInvocationOperation)context.Operation;
+            if (invocationOperation.Arguments.Length < 1)
+            {
+                return;
+            }
+
+            // Check that the object is a StringBuilder
+            if (!context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemTextStringBuilder, out INamedTypeSymbol? stringBuilderType))
+            {
+                return;
+            }
+
+            IMethodSymbol appendStringMethod = stringBuilderType
+                .GetMembers("Append")
+                .OfType<IMethodSymbol>()
+                .WhereAsArray(s =>
+                    s.Parameters.Length == 1 &&
+                    s.Parameters[0].Type.SpecialType == SpecialType.System_String)
+                .FirstOrDefault();
+            if (appendStringMethod is null)
+            {
+                return;
+            }
+
+            if (!invocationOperation.TargetMethod.Equals(appendStringMethod))
+            {
+                return;
+            }
+
+            ImmutableArray<IArgumentOperation> arguments = invocationOperation.Arguments;
+            IArgumentOperation firstArgument = arguments[0];
+
+            // We don't handle class fields. Only local declarations
+            if (!(firstArgument.Value is ILocalReferenceOperation argumentValue))
+            {
+                return;
+            }
+
+            ILocalSymbol localArgumentDeclaration = argumentValue.Local;
+
+            // If there are multiple declarations, bail
+            var semanticModel = context.Operation.SemanticModel;
+            var cancellationToken = context.CancellationToken;
+            SyntaxReference declaringSyntaxReference = localArgumentDeclaration.DeclaringSyntaxReferences.First();
```

Comment:
Is this guaranteed to always find one, or might First throw?

Reply:
I can't think of a scenario where it is null, but I changed it to FIrstOrDefault and checking for null now just to be safe.